### PR TITLE
feat: [HL-18] Add and configure Xcode template 🔨

### PR DESCRIPTION
**Pull request by Ahattalla**
### Description:

This pull request for adding MVVM template to Xcode.

### Changes made:

- Added MVVM template to MVVM-Template directory in project root
- Added a makefile to automatically add the template to Xcode (commands for installing and uninstalling template is in the readme file)

### CODE REVIEW

File: MVVM-Template/Makefile [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Makefile
+
+# Define color codes
+ORANGE = \033[0;33m
+GREEN = \033[0;32m
+CRIMSON = \033[0;31m
+BLUE = \033[0;34m
+CYAN = \033[0;36m
+YELLOW = \033[0;33m
+RESET = \033[0m
```
Comment (ahmdmhasn) [inline MVVM-Template/Makefile]:
```suggestion
GREEN = \033[0;32m
RESET = \033[0m
```

File: MVVM-Template/Makefile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	@sleep 0.4
+	@printf "\r$(GREEN)Installing template....."
+	@sleep 0.4
+	@printf "\r$(CYAN)Installing template......"
+	@sleep 0.4
+	@printf "\r$(RESET)Installing template......."
+	@sleep 0.4
+	@printf "\r$(GREEN)Installing template........\n"
+	@$(MAKE) install_templates
+
```
Comment (ahmdmhasn) [inline MVVM-Template/Makefile]:
```suggestion
```

No dead code policy. Any added code should have value. I like the colors btw 😄❤️

File: MVVM-Template/Makefile [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+XCODE_USER_TEMPLATES_DIR=~/Library/Developer/Xcode/Templates/File\ Templates
+TEMPLATES_DIR=MVVM
```
Comment (ahmdmhasn) [inline MVVM-Template/Makefile]:
```suggestion
TEMPLATES_DIR=Templates
```

Let's rename the folder to `Templates`

File: .swiftlint.yml [hunk 4]
```diff
@@ -35,6 +35,7 @@ analyzer_rules:
 excluded:
   - Carthage
   - Pods
+  - MVVM-Template
```
Comment (ahmdmhasn) [inline .swiftlint.yml]:
```suggestion
  - Templates
```

**Review by ahmdmhasn — APPROVED** [review]
Great effort! 🚀

**Review by abdelhamid-f-nasser — APPROVED** [review]
This will make it way easier for all us throughout the project !! 💪🏽🔥
Great work ^_^

**Pull request merged**